api/v1beta1/watcherapi_types.go

```diff
@@ -17,25 +17,28 @@ limitations under the License.
 package v1beta1
 
 import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // WatcherAPISpec defines the desired state of WatcherAPI
 type WatcherAPISpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of WatcherAPI. Edit watcherapi_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	WatcherCommon `json:",inline"`
```

Comment:
so, first of all, I believe you are commenting on the lack of 
```
ContainerImage string `json:"containerImage"` 
```

in this proposal right? i.e. the fact that we don't carry it here (because this is now) set by OpenStackVersion)
I think your proposal about avoiding CRD churn is reasonable, but not sure if it would be 'unexpected' for anyone that would go looking for the containerImage on the crd (though as you said, eventually we'd want to remove that one too?)

Reply:
I purposely left this parameter option out until it could be used (when we implement the deployment) in hopes of having PRs as small as possible. @amoralej has a good point about CI, we might want to use a different container image than the one in the bundle

Reply:
so some context

all the lower level crd used to have a required

ContainerImage string `json:"containerImage"` 

filed that was set via the upper layers

but when we created the OpenStack version CRD  we moved the URL form the individual CRDs to that common one.

since we don't have that integration for watcher we have 2 choices.

either we only implement the other two levels of precedence

hardcoded fallback to quey.io:latest when an environment variable is not set (vai the bundle)

or we can temporary add ContainerImage string `json:"containerImage"` 

I'm leang toward defering this until we add support for the container (stateful set) creation 

we can choose if we want to only support the first 2 levels fo precendce and defer the configibale URL until we have integration with the OpenStack Version CRD
or we can add ContainerImage string `json:"containerImage"` as an interim solution.